drivers/spi/spi-intel.c

```diff
@@ -1113,12 +856,19 @@ static int intel_spi_init(struct intel_spi *ispi)
 		return -EINVAL;
 	}
 
+	if ((ispi->swseq_erase || !erase_64k) && !ispi->swseq_enabled)
+	{
+		dev_err(ispi->dev, "software sequencer not enabled and erase"
+				   "is not supported by hardware sequencing\n");
+		return -EINVAL;
+	}
+
 	/*
 	 * Some controllers can only do basic operations using hardware
 	 * sequencer. All other operations are supposed to be carried out
 	 * using software sequencer.
 	 */
-	if (ispi->swseq_reg) {
+	if (ispi->swseq_reg && ispi->swseq_enabled) {
 		/* Disable #SMI generation from SW sequencer */
```

Comment:
in this and in the following change you are keeping the SW sequencing code here.
Whereas for example in mem_op_supported_on_spi_locked you moved the function outside to the sw sequencing code. I think we should be consistent, whether to keep the code and protect it with swseq_enabled or use the functions + swseq_enabled